api/src/main/java/marquez/db/EventDao.java

```diff
@@ -0,0 +1,37 @@
+package marquez.db;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.List;
+import marquez.db.mappers.RawLineageEventMapper;
+import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+
+@RegisterRowMapper(RawLineageEventMapper.class)
+public interface EventDao extends BaseDao {
+
+  @SqlQuery(
+      """
+    SELECT event
+    FROM lineage_events
+    ORDER BY event_time DESC
+    LIMIT :limit
+    OFFSET :offset""")
+  List<JsonNode> getAll(int limit, int offset);
+
+  /**
+   * This is a "hack" to get inputs/outputs namespace from jsonb column: <a
+   * href="https://github.com/jdbi/jdbi/issues/1510#issuecomment-485423083">explanation</a>
+   */
+  @SqlQuery(
+      """
+    SELECT le.event
+    FROM lineage_events le, jsonb_array_elements(coalesce(le.event -> 'inputs', '[]'::jsonb) || coalesce(le.event -> 'outputs', '[]'::jsonb)) AS ds
+    WHERE le.job_namespace = :namespace
+    OR ds ->> 'namespace' =  :namespace
+    ORDER BY event_time DESC
+    LIMIT :limit
+    OFFSET :offset""")
+  List<JsonNode> getByNamespace(
+      @Bind("namespace") String namespace, @Bind("limit") int limit, @Bind("offset") int offset);
+}
```

Comment:
This looks like it would be expensive on a big table. I would check the perf difference of filtering on the columns you added indices for vs this.

Reply:
With @wslulciuc changes described in https://github.com/MarquezProject/marquez/pull/2076 the json based filter would be replaced with fast join. 

It would be possible to [create jsonb index](https://www.postgresql.org/docs/current/datatype-json.html#JSON-INDEXING) - however, I think it would be overkill.